packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -133,13 +502,77 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+            /* @conditional-compile-remove(mention) */
+            handleOnChange(e, newValue);
+            /* @conditional-compile-remove(mention) */
+            return;
+            onChange(e, newValue);
+          }}
+          /* @conditional-compile-remove(mention) */
+          onSelect={(e) => {
+            if (caretIndex !== undefined) {
+              e.currentTarget.setSelectionRange(caretIndex, caretIndex);
+              setCaretIndex(undefined);
+              return;
+            }
+            //TODO: need to check to navigate before/after space correctly in tag + when selecting by mouse
+            /* @conditional-compile-remove(mention) */
+            if (
+              shouldHandleOnMouseDownDuringSelect &&
+              e.currentTarget.selectionStart !== null &&
+              e.currentTarget.selectionStart === e.currentTarget.selectionEnd
+            ) {
+              // handle mention click
+              const mentionTag = findMentionTagForSelection(tagsValue, e.currentTarget.selectionStart);
+              if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
+                if (e.currentTarget.selectionDirection === null) {
+                  e.currentTarget.setSelectionRange(
+                    mentionTag.plainTextBeginIndex,
+                    mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex
+                  );
+                } else {
+                  e.currentTarget.setSelectionRange(
+                    mentionTag.plainTextBeginIndex,
+                    mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex,
+                    e.currentTarget.selectionDirection
+                  );
+                }
+                setSelectionStartValue(mentionTag.plainTextBeginIndex);
+                setSelectionEndValue(mentionTag.plainTextEndIndex ?? mentionTag.plainTextBeginIndex);
+              } else {
+                setSelectionStartValue(e.currentTarget.selectionStart);
+                setSelectionEndValue(e.currentTarget.selectionEnd);
+              }
+            } else {
+              updateSelectionIndexesWithMentionIfNeeded(e);
+            }
+            /* @conditional-compile-remove(mention) */
+            setShouldHandleOnMouseDownDuringSelect(false);
+          }}
+          onMouseDown={() => {
+            // as events order is onMouseDown -> onSelect -> onClick
+            // onClick and onMouseDown can't handle clicking on mention event because
+            // onClick has wrong range as it's called after onSelect
+            // onMouseDown doesn't have correct selectionRange yet
+            // so we need to handle onMouseDown to prevent onSelect default behavior
+            /* @conditional-compile-remove(mention) */
+            setShouldHandleOnMouseDownDuringSelect(true);
+          }}
+          onTouchStart={() => {
+            // see onMouseDown for more details
+            /* @conditional-compile-remove(mention) */
```

Comment:
Move /* @conditional-compile-remove(mention) */ to the top of onTouchStart

Reply:
Added to refactor task